DataFormats/Detectors/FIT/FV0/include/DataFormatsFV0/RecPoints.h

```diff
@@ -0,0 +1,104 @@
+// or submit itself to any jurisdiction.
+
+/// \file RecPoints.h
+/// \brief Definition of the FV0 RecPoints class
+
+#ifndef ALICEO2_FV0_RECPOINTS_H
+#define ALICEO2_FV0_RECPOINTS_H
+
+#include "CommonDataFormat/InteractionRecord.h"
+#include "CommonDataFormat/TimeStamp.h"
+#include "DataFormatsFV0/ChannelData.h"
+#include "CommonDataFormat/RangeReference.h"
+#include "DataFormatsFV0/BCData.h"
+#include <array>
+#include "Rtypes.h"
+#include <TObject.h>
+#include <gsl/span>
+
+namespace o2
+{
+namespace fv0
+{
+struct ChannelDataFloat {
+
+  int channel = -1;       // channel Id
+  double time = -20000;   // time in ps, 0 at the LHC clk center
+  double charge = -20000; // charge [channels]
+  int adcId = -1;         // QTC chain
+
+  ChannelDataFloat() = default;
+  ChannelDataFloat(int Channel, double Time, double Charge, int AdcId)
+  {
+    channel = Channel;
+    time = Time;
+    charge = Charge;
+    adcId = AdcId;
+  }
+
+  void print() const;
+
+  ClassDefNV(ChannelDataFloat, 1);
+};
+
+class RecPoints
+{
+
+ public:
+  enum TimeTypeIndex : int { TimeFirst,
+                             TimeGlobalMean,
+                             TimeSelectedMean };
+
+  o2::dataformats::RangeReference<int, int> ref;
+  o2::InteractionRecord mIntRecord;
+  RecPoints() = default;
+  RecPoints(const std::array<Float_t, 3>& collisiontime,
+            int first, int ne, o2::InteractionRecord iRec, o2::fv0::Triggers triggers)
+    : mCollisionTime(collisiontime)
+  {
+    ref.setFirstEntry(first);
+    ref.setEntries(ne);
+    mIntRecord = iRec;
+    mTriggers = triggers;
+  }
+  ~RecPoints() = default;
+
+  void print() const;
+
+  o2::fv0::Triggers mTriggers; // pattern of triggers  in this BC
+
+  float getCollisionTime(TimeTypeIndex type) const { return mCollisionTime[type]; }
+  float getCollisionFirstTime() const { return getCollisionTime(TimeFirst); }
+  float getCollisionGlobalMeanTime() const { return getCollisionTime(TimeGlobalMean); }
+  float getCollisionSelectedMeanTime() const { return getCollisionTime(TimeSelectedMean); }
+  bool isValidTime(TimeTypeIndex type) const { return getCollisionTime(type) < o2::InteractionRecord::DummyTime; }
+  void setCollisionTime(Float_t time, TimeTypeIndex type) { mCollisionTime[type] = time; }
+
+  o2::fv0::Triggers getTrigger() const { return mTriggers; }
+
+  o2::InteractionRecord getInteractionRecord() const { return mIntRecord; };
+
+  void SetMgrEventTime(Double_t time) { mTimeStamp = time; }
+
+  gsl::span<const ChannelDataFloat> getBunchChannelData(const gsl::span<const ChannelDataFloat> tfdata) const;
+
+ private:
+  std::array<Float_t, 3> mCollisionTime = {2 * o2::InteractionRecord::DummyTime,
+                                           2 * o2::InteractionRecord::DummyTime,
+                                           2 * o2::InteractionRecord::DummyTime};
+  Double_t mTimeStamp = 2 * o2::InteractionRecord::DummyTime; //event time from Fair for continuous
```

Comment:
This seems never used, can it be suppressed (@AllaMaevskaya also in FT0)

Reply:
Done